pandas/core/indexes/timedeltas.py

```diff
@@ -180,8 +194,18 @@ def __new__(cls, data=None, unit=None, freq=None, start=None, end=None,
         return subarr
 
     @classmethod
-    def _simple_new(cls, values, name=None, freq=None, **kwargs):
-        result = super(TimedeltaIndex, cls)._simple_new(values, freq, **kwargs)
+    def _simple_new(cls, values, name=None, freq=None, dtype=_TD_DTYPE):
+
+        # `dtype` is not always passed, but if it is, it should always
+        #  be m8[ns]
+        assert dtype == _TD_DTYPE
+
+        assert isinstance(values, np.ndarray), type(values)
+        if values.dtype == 'i8':
+            values = values.view('m8[ns]')
+        assert values.dtype == 'm8[ns]', values.dtype
```

Comment:
Two points:

* Generally not a big fan of bare `assert` like these, unless they're internal.  Are these user-facing in any way?
* Even if they're internal are these `assert` statements tested?

Reply:
Largely these got added for debugging and left them in since they behave like especially-emphatic comments.  On the next pass I'll make sure that they only go in private methods.